frontend/src/pages/operatorPage/OperatorPage.js

```diff
@@ -81,9 +68,18 @@ class OperatorPage extends React.Component {
     }
   };
 
-  updateVersion = operator => {
-    this.setState({ operator });
-    this.props.history.push(`/operator/${operator.name}`);
+  updateChannel = channel => {
+    const { operator } = this.props;
+
+    if (channel.name !== operator.channel) {
+      this.props.history.push(`/operator/${channel.name}/${channel.currentCSV}`);
+    }
+  };
+
+  updateVersion = version => {
+    const { operator } = this.props;
+
+    this.props.history.push(`/operator/${operator.channel}/${version.name}`);
```

Comment:
Can you have multiple versions per channel?
Side Note: Should this be prefixed with `operator.name`?

Reply:
or is version name the operator name?

Reply:
The version object has both the operator's metadata.name and the version string. So version.name is correct here.